src/main/java/core/basesyntax/service/operation/SupplyOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
```

The apply(...) method does not validate its input. Add checks for `transaction != null`, non-null/non-empty `transaction.getFruit()`, and `transaction.getQuantity() >= 0`. Per the checklist: "Quantity must be a non-negative integer; invalid quantity should cause a RuntimeException." Throw informative RuntimeException messages on invalid inputs to fail fast and avoid corrupting Storage.